### PR TITLE
Upgrade com.github.johnrengelman:shadow to 8.1.1

### PR DESCRIPTION
**Pull request by Phone-Guru**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -72,15 +72,18 @@ plugins {
...
-    repositories {
-        gradlePluginPortal()
-        google()
-    }
-    dependencies {
-        classpath("gradle.plugin.com.github.johnrengelman:shadow:8.0.0")
+  repositories {
+    maven {
+      url = uri("https://plugins.gradle.org/m2/")
     }
```
Comment (iBotPeaches) [inline build.gradle.kts:78]: Isn't this the same as `gradlePluginPortal`?

File: build.gradle.kts [hunk 2]
```diff
@@ -72,15 +72,18 @@ plugins {
...
+  }
+  dependencies {
+    classpath("com.github.johnrengelman:shadow:8.1.1")
+  }
 }
 
+apply(plugin = "com.github.johnrengelman.shadow")
```
Comment (iBotPeaches) [inline build.gradle.kts:85]: Do you need to apply this? If it was attached to build script classpath its already there for building.

**Review by iBotPeaches — CHANGES_REQUESTED** [review]
Looks like we lost the 4 space indent, these new changes are 2 indent.

File: build.gradle.kts (hunk 1 above)
Comment (Phone-Guru) [reply build.gradle.kts:78]: This is how it was written at https://plugins.gradle.org/plugin/com.github.johnrengelman.shadow

File: build.gradle.kts (hunk 2 above)
Comment (Phone-Guru) [reply build.gradle.kts:85]: again, this is how it was at https://plugins.gradle.org/plugin/com.github.johnrengelman.shadow

Comment (iBotPeaches) [reply build.gradle.kts:85]: Looks like that says two different methods though and you have both, so just a bit lost.

**Comment by iBotPeaches** [comment]
@Phone-Guru - Thanks for spotting this out of date version, but I went ahead and did what I was trying to explain. Its tough to follow online docs because they don't really fully show all the ways a plugin works. They don't show buildscript vs runtime dependencies and suffer in groovy vs kotlin.

So we wouldn't really go anywhere just taking the examples online. Unfortunately Gradle has a mix of a lot of old/new docs around.

So take a look at this diff if you are curious what I did - https://github.com/iBotPeaches/Apktool/pull/3259

**Pull request closed without merging**